packages/cyberstorm/src/components/Button/Button.tsx

```diff
@@ -79,17 +83,21 @@ const Button = React.forwardRef<
     const fRef = forwardedRef as React.ForwardedRef<HTMLButtonElement>;
```

Comment:
There should be 4 different outcomes. 1. Plain button without tooltip 2. Button (non-plain) without tooltip 3. Plain button with tooltip 4. Button (non-plain) with tooltip. You can check if the `tooltipText` is `null` `if (tooltipText) {` and then put the `if (plain)` checks inside of the `tooltipText` if check and the following `else` block.

Reply:
To take this idea further, the code duplication in if-else-blocks can be reduced with something like this: 

```jsx
interface TooltipWrapperProps extends PropsWithChildren {
  tooltipText?: string;
}

const TooltipWrapper = (props: TooltipWrapperProps) =>
  props.tooltipText ? (
    <Tooltip content={props.tooltipText} side="bottom">
      {props.children}
    </Tooltip>
  ) : (
    <>{props.children}</>
  );
```

and then using it:

```jsx
  if (plain) {
    const fRef = forwardedRef as React.ForwardedRef<HTMLDivElement>;
    const ref = fRef || fallbackRef;
    return (
      <TooltipWrapper tooltipText={type}>
        <div
          {...forwardedProps}
          ref={ref}
          className={`${styles.root} ${getIconAlignment(
            iconAlignment
          )} ${getStyle(colorScheme)} ${getPaddingSize(paddingSize)}`}
        >
          {children}
        </div>
      </TooltipWrapper>
    );
  } else {
    const fRef = forwardedRef as React.ForwardedRef<HTMLButtonElement>;
    const ref = fRef || fallbackRef;
    return (
      <TooltipWrapper tooltipText={type}>
        <button
          {...forwardedProps}
          ref={ref}
          type={type}
          className={`${styles.root} ${getIconAlignment(
            iconAlignment
          )} ${getStyle(colorScheme)} ${getPaddingSize(paddingSize)}`}
          onClick={onClick}
        >
          {children}
        </button>
      </TooltipWrapper>
    );
  }
```

As I'm just yelling from the sidelines, use your own judgement if this approach makes sense here or if it's overengineering or whatever. It would probably be too daring (and outside the scope) to use the same conditional rendering approach to regular/plain button parts?

Reply:
Thanks! If-else stacking was most likely way I would have approached this and even it felt uncomfortable way to do so finding this way to do it would have taken bit more time.. 
Sadly overengineering is like second habit of mine, so I'll let @Oksamies make executive decision on this one!